app/main.py

```diff
@@ -1,13 +1,47 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8",
+            }
+            for error in errors
+        ],
+        "path": file_path,
+        "status": "failed" if len(errors) != 0
+        else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                {
+                    "line": value["line_number"],
+                    "column": value["column_number"],
+                    "message": value["text"],
+                    "name": value["code"],
+                    "source": "flake8",
+                }
+                for value in values
+            ],
+            "path": errors,
+            "status": "failed" if len(values) != 0
+            else "passed"
```

Comment:
You are repeating yourself here as well (twice). You created functions above to use them

Reply:
I have remove repeating